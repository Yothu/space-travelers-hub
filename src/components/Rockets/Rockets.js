import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import { getRocketsFromAPI, listRockets } from '../../redux/rockets/rockets';
import Rocket from '../Rocket/Rocket';

const RocketsContainer = styled.div`
  padding: 1rem;
`;

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocketsReducer);

  useEffect(async () => {
    const APIrockets = await getRocketsFromAPI();
    dispatch(listRockets(APIrockets));
  }, [dispatch]);

  return (
    <RocketsContainer>
      {
        rockets.map((rocket) => (
          <Rocket
            id={rocket.id}
            key={rocket.id}
            name={rocket.name}
            image={rocket.image}
            description={rocket.description}
          />
        ))
      }
    </RocketsContainer>
  );
};

export default Rockets;
