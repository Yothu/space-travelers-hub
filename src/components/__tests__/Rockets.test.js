import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../Rockets/Rockets';
import store from '../../redux/configureStore';

describe('Rockets page tests', () => {
  it('it renders', () => {
    const rockets = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(rockets).toMatchSnapshot();
  });
});
