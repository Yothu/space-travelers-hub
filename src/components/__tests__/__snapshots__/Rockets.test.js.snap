// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets page tests it renders 1`] = `
<div
  className="sc-iCfMLu nCdzM"
/>
`;
